service/matching/matching_engine.go

```diff
@@ -958,55 +958,78 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 			defaultBuildId := getDefaultBuildId(userData.GetVersioningData().GetAssignmentRules())
 			req.Versions = &taskqueuepb.TaskQueueVersionSelection{BuildIds: []string{defaultBuildId}}
 		}
-		// collect internal info
+		rootPM, _, err := e.getTaskQueuePartitionManager(ctx, rootPartition, true, loadCauseDescribe)
+		if err != nil {
+			return nil, err
+		}
+		rootPMImpl, ok := rootPM.(*taskQueuePartitionManagerImpl)
+		if !ok {
+			return nil, fmt.Errorf("failed type assertion on root partition")
```

Comment:
I think the convention that the linter is trying to enforce here is to use `serviceerror` instead.

I think the applicable one would be `serviceerror.NewInternal("failed type assertion on root partition")`

Reply:
We just talked at standup the other day about whether to follow http or grpc convention on "internal" errors and whether or not they're retryable. I think the conclusion was that we're following grpc convention which says they are serious and therefore our grpc settings don't retry on internal error. I think that applies here, since we really really don't expect this assertion to fail.

See grpc status code descriptions for more info: https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
Having said that, I actually think that instead of casting the interface to the implementation so that you can use the private `timeSinceLastFanOut` method, you should just add `TimeSinceLastDescribeTaskQueueFanOut` to the `taskQueuePartitionManager` interface. 

This is necessary information you need to get from the partition, so I think it's justified to add to the interface. Reaching through the interface via the type assertion breaks the interface api boundary which should not be done if it can be solved by adding to the interface.

Reply:
Having read even further, I see that you use the rootPMImpl to load and store the cache as well. I would support a follow-up PR that removes the need to break the interface abstraction, but for this PR I'm ok leaving it as is.